app/Policies/PostPolicy.php

```diff
@@ -71,7 +71,7 @@ public function delete(User $user, Post $post)
     {
         //
 
-        return $user->id == $post->user_id || $user->role_id == 2;
```

Comment:
if im gonna work with you on that project i don't know what  1 and 2 mean and also if you are working on a big project you won't remember that `1` means admin and `2` means supperAdmin and so on..  so instead of using ints in your code you can use connstant or enums. check those links [https://laravel.com/docs/9.x/validation#rule-enum](url) [https://www.w3schools.com/php/php_constants.asp#:~:text=A%20constant%20is%20an%20identifier,sign%20before%20the%20constant%20name).](url)